scripts/e2e/e2e.sh

```diff
@@ -157,7 +161,7 @@ start_docker() {
 on_exit() {
   [ "${VM_CREATED}" ] || return 0
   get_bootstrap_vm "${CONTEXT}"
-  delete_vm "${TARGET_VM_PRE}"
+  delete_vm "${CLUSTER_NAME}"
```

Comment:
I like how this works a lot better -- and confirmed it cleans up correctly on VMC.

I'll leave it up to you if you want to leave this as-is and file a follow-up PR after this merges, or make it part of tihs PR, but since we are no longer using `TARGET_VM_PRE`, we could do *a lot* more clean up. Basically, remove all mentions of it.

That means removing it from:
```
~/git/sigs.k8s.io/cluster-api-provider-vsphere (master)]$ git grep TARGET_VM_PRE
scripts/e2e/bootstrap_job.template:          - name: TARGET_VM_PREFIX
scripts/e2e/bootstrap_job/clusterctl.sh:# TARGET_VM_PREFIX
scripts/e2e/bootstrap_job/spec/machines.template:    generateName: ${TARGET_VM_PREFIX}-
scripts/e2e/bootstrap_secret.template:  target-vm-prefix: ${TARGET_VM_PREFIX}
scripts/e2e/e2e.sh:# TARGET_VM_PRE contains a string to uniquely indentify the newly created VM(s)
scripts/e2e/e2e.sh:TARGET_VM_PRE=
scripts/e2e/e2e.sh:  delete_vm "${TARGET_VM_PRE}"
scripts/e2e/e2e.sh:TARGET_VM_PRE="clusterapi-""${RANDOM_STR}"
scripts/e2e/e2e.sh:export_base64_value "TARGET_VM_PREFIX" "${TARGET_VM_PRE}"
```

Also, we can remove the `RANDOM_STR` variable from `e2e.sh` and the function that populates it. We don't need it at all. Good work!